HashTable.java

```diff
@@ -0,0 +1,192 @@
+import java.util.*;
+import java.lang.reflect.Array;
+
+class HashTable <K, V> {
+
+    // The array cannot be smaller than MIN_CAPACITY at any time
+    private static final int MIN_CAPACITY = 32;
+    private int approxSize = Integer.highestOneBit(MIN_CAPACITY) << 1;
```

Comment:
Avoid setting the value for non static variables outside of the default constructor.